render/render.go

```diff
@@ -142,16 +227,33 @@ func edgeMetadata(t report.Topology, mapFunc MapFunc, srcRenderableID, dstRender
 			continue
 		}
 		if src != report.TheInternet {
-			mapped, _ := mapFunc(t.NodeMetadatas[src])
+			mapped, _ := m.Mapper(t.NodeMetadatas[src])
 			src = mapped.ID
 		}
 		if dst != report.TheInternet {
-			mapped, _ := mapFunc(t.NodeMetadatas[dst])
+			mapped, _ := m.Mapper(t.NodeMetadatas[dst])
 			dst = mapped.ID
 		}
 		if src == srcRenderableID && dst == dstRenderableID {
 			metadata.Flatten(edgeMeta)
 		}
 	}
-	return metadata
+	return metadata.Transform()
+}
+
+// Render produces a set of RenderableNodes given a Report
+func (f FilterUnconnected) Render(rpt report.Report) RenderableNodes {
+	input := f.Renderer.Render(rpt)
+	output := RenderableNodes{}
+	for id, node := range input {
+		if len(node.Adjacency) == 0 {
+			continue
+		}
+
+		output[id] = node
+		for _, id := range node.Adjacency {
+			output[id] = input[id]
+		}
+	}
+	return output
 }
```

Comment:
I'm not *totally* sure what the second clause in your loop does (walking each node.Adjacency) but I think this is equivalent:

```go
output := f.Renderer.Render(rpt)
for id, node := range output {
    if len(node.Adjacency) == 0 {
        delete(output, id)
    }
}
return output
```